added pong object (control version of pong~)

porres opened the pull request:
this update and object inclusion is actually from a third party: Derek Kwan

here's his pull request to my cyclone github fork (https://github.com/porres/pd-cyclone)

"I've added source code for a control rate version of pong. It is found at hammer/pong.c. I edited the pong~ help file to create pong's help file (but made no changes to pong~-help.pd). In the help file, I declared the object as cyclone/pong but I haven't built it that way on my computer (didn't make a library for the single object). I also added a line to the Makefile under hammer classes, copying the convention of the other lines, but I'm not quite sure how that makefile works so I don't know if this is sufficient or if I need to add more (I tested pong on my computer by using the template from puredata.info). I've attached a picture demonstrating tests of my pong control rate object."

![](https://cloud.githubusercontent.com/assets/7247906/13072051/0fb4e404-d44c-11e5-8486-7aa2ade64403.png)
![unnamed](https://cloud.githubusercontent.com/assets/5360142/13080234/eda49e82-d4ae-11e5-9765-797afaadd677.png)

porres commented:
I was able to compile the library and verify that the [pong] object did build, and that it seems to be working properly in my tests.

I accidentally inserted in this pull the add of atodb and atodb~ help corresponding to the c files that Amurtet uploaded

umlaeute commented:
@porres if your pull-request (PR) accidentally includes commits that you don't want to be in *this* PR, you can simply close this PR and create a new one.

also you probably should read about how Pull Requests work and how they relate to *git branches*.
if you create a PR from *your* `master` branch then the PR will include *everything* in your branch since it diverged from the original repository.
if you add new features to *that very* master branch afterwards and push them, they will *also* show up in the Pull Request (this is well documented; you are even reminded of this whenever you created a new pull request).

porres committed:
Update README.md

porres committed:
Update README.md

porres committed:
Update README.md

porres committed:
Update README.md

porres committed:
Update README.md

porres committed:
Update README.md

porres committed:
Update README.md

porres committed:
Update README.md

porres committed:
Update README.md

porres committed:
Update README.md

derekxkwan committed:
adding updated round~ and new round

derekxkwan committed:
adding updated round~

derekxkwan committed:
updating makefile

derekxkwan committed:
updating makefile

porres committed:
Merge pull request #3 from derekxkwan/master

adding updated round~ and new round, some changes to makefile

porres committed:
Update README.md

porres committed:
Update README.md

derekxkwan committed:
updating round, round~, pong

derekxkwan committed:
updating round, round~, pong

porres committed:
Merge pull request #4 from derekxkwan/master

updating pong, round, round~

derekxkwan committed:
fixing pong, round, round~

derekxkwan committed:
fixing pong where minval = maxval

porres committed:
Merge pull request #5 from derekxkwan/master

fixing pong, round, round~

derekxkwan committed:
adding forgotten statics to a couple of pong functions

derekxkwan committed:
changing pong's mode numberings to be consistent with pong~

derekxkwan committed:
fixing pong, adding brand new pong~ code

derekxkwan committed:
fixing Makefile for new pong~

derekxkwan committed:
fixing mode none bug in pong~

porres committed:
Merge pull request #6 from derekxkwan/master

fixing pong (more), adding new code for pong~, changing Makefile

derekxkwan committed:
updating pong, fixing pong~

porres committed:
Merge pull request #7 from derekxkwan/master

updating pong, fixing pong~

jwmatthys committed:
rewrote train~ to tick on width=0 or width=1

jwmatthys committed:
rename variable in2val to width_inlet for clarity

jwmatthys committed:
no automatic bang on init

porres committed:
Merge pull request #8 from jwmatthys/master

train~ rewrite

derekxkwan committed:
fixing pong/pong~

porres committed:
Merge pull request #9 from derekxkwan/master

fixing pong/pong~

derekxkwan committed:
changing some color defaults and adding an inlet to scope~

porres committed:
Merge pull request #10 from derekxkwan/master

changing some color defaults and adding an inlet to scope~

derekxkwan committed:
removing wrong way to make new unneeded inlet, changing select color

derekxkwan committed:
changing DEFGRcolors

porres committed:
Merge pull request #11 from derekxkwan/master

removing wrong way to make new unneeded inlet, changing select color

d-i-s committed:
Added files via upload

porres committed:
Merge pull request #12 from amurtet/master

Added files via upload

porres committed:
Revert "Added files via upload"

porres committed:
Merge pull request #13 from porres/revert-12-master

Revert "Added files via upload"

d-i-s committed:
Add atodb

Add atodb control version.

d-i-s committed:
Add dbtoa

Add dbtoa control version.

d-i-s committed:
Add atodb~ & dbtoa~

Add atodb~ and dbtoa~ audio version.

d-i-s committed:
Add scale

Add scale to hammer

d-i-s committed:
Update Makefile

Update Makefile with 5 new objs.

d-i-s committed:
Update and rename atodb~.c to atodb.c

d-i-s committed:
Update and rename dbtoa~.c to dbtoa.c

porres committed:
Merge pull request #15 from amurtet/amurtet-obj-1

Add dbtoa

porres committed:
Merge pull request #16 from amurtet/amurtet-obj-3

Amurtet obj 3

porres committed:
Merge pull request #17 from amurtet/amurtet-obj-2

Amurtet obj 2

porres committed:
Merge pull request #18 from amurtet/amurtet-obj

Add atodb

porres committed:
Revert "Amurtet obj 3"

porres committed:
Merge pull request #19 from porres/revert-16-amurtet-obj-3

Revert "Amurtet obj 3"

porres committed:
Revert "Add dbtoa"

porres committed:
Merge pull request #20 from porres/revert-15-amurtet-obj-1

Revert "Add dbtoa"

porres committed:
Revert "Amurtet obj 2"

porres committed:
Merge pull request #21 from porres/revert-17-amurtet-obj-2

Revert "Amurtet obj 2"

porres committed:
Revert "Add atodb"

porres committed:
Merge pull request #22 from porres/revert-18-amurtet-obj

Revert "Add atodb"

porres committed:
Revert "Revert "Add dbtoa""

porres committed:
Merge pull request #23 from porres/revert-20-revert-15-amurtet-obj-1

Revert "Revert "Add dbtoa""

porres committed:
Revert "Revert "Amurtet obj 2""

porres committed:
Merge pull request #24 from porres/revert-21-revert-17-amurtet-obj-2

Revert "Revert "Amurtet obj 2""

porres committed:
Revert "Revert "Add atodb""

porres committed:
Merge pull request #25 from porres/revert-22-revert-18-amurtet-obj

Revert "Revert "Add atodb""

porres committed:
Revert "Revert "Amurtet obj 3""

porres committed:
Merge pull request #26 from porres/revert-19-revert-16-amurtet-obj-3

Revert "Revert "Amurtet obj 3""

porres committed:
Update README.md

porres committed:
Update README.md

porres committed:
Update README.md

porres committed:
Update README.md

derekxkwan committed:
updating scope~

derekxkwan committed:
updating scope~

porres committed:
Merge pull request #28 from derekxkwan/master

updating scope~

derekxkwan committed:
adding back args in scope~

porres committed:
Merge pull request #29 from derekxkwan/master

adding back args in scope~

porres committed:
Update buffir.c

porres committed:
Update buffir.c

revert maxsize as it didn't work

derekxkwan committed:
adding more args to scope~

porres committed:
Merge pull request #30 from derekxkwan/master

adding more args to scope~

derekxkwan committed:
fixing save state of scope~

derekxkwan committed:
changing selcolor back in scope~

derekxkwan committed:
changing color of select border in scope~

porres committed:
Merge pull request #31 from derekxkwan/master

fixing save state of scope~

porres committed:
Update Scope.c

porres committed:
update select border

border select color that seems to be in max: 5AADEF

porres committed:
fix comment

porres committed:
including help files for the new objects

help files of atodb/atodb~/dbtoa/dbtoa~/pong/round/round~

porres committed:
including help file for the updated object pong~

pong~ is now updated to Max 7, this is the help file

porres committed:
new help file of the updated file scope~

scope~ is now updated to max 7, this is the help file

porres committed:
new help files for the MSP/sickle objects

help files are related to Max 7, many of the objects are actually bugged and do not work as they should and as described in the help files

porres committed:
new trunc~ object code

porres committed:
help for the new object trunc~

porres committed:
updating make file to include trunc~

derekxkwan committed:
fixing selected scope~ background

porres committed:
Merge pull request #32 from derekxkwan/master

fixing selected scope~ background

porres committed:
updated scope~ help

porres committed:
upfdate help files to explain list input

porres committed:
code for new loadmess object

porres committed:
update make to include loadmess object

porres committed:
help file for new loadmess object

porres committed:
Update README.md

porres committed:
Update README.md

porres committed:
Update README.md

porres committed:
Update README.md

porres committed:
an update

porres committed:
new teeth~ abstraction help

porres committed:
deletar essa porra

caralho

porres committed:
add teeth~ abstraction

derekxkwan committed:
updating bitxor~, curve~, deltaclip~

porres committed:
Merge pull request #33 from derekxkwan/master

updating bitxor~, curve~, deltaclip~

porres committed:
new help for updated bitxor~ (now compatible to Max7)

porres committed:
new help correcting "click" instead of double click

double click should be implemented though

porres committed:
changing name

porres committed:
new thresh~ abstraction and help

porres committed:
update help for thresh~  abstraction

porres committed:
change file name

porres committed:
new downsamp~ abstraction and help file

porres committed:
new help file for downsamp~

porres committed:
and again

porres committed:
new degrade~ abstraction and help

porres committed:
trying nettles update

porres committed:
updating nettles help symbol for ==~ and including equals~ abs

porres committed:
updating ==~ help and abstractions

porres committed:
more updates to ==~ abstractions

porres committed:
again updating ==~ abstractions and help

porres committed:
nettles help update

porres committed:
updating ==~ and creating a new object equals~

derekxkwan committed:
updating trapezoid~

porres committed:
Merge pull request #34 from derekxkwan/master

updating trapezoid~

porres committed:
nettles again

derekxkwan committed:
fixing typo in trapezoid_hi

porres committed:
Merge pull request #35 from derekxkwan/master

fixing typo in trapezoid_hi

derekxkwan committed:
trivial commit to trapezoid~

porres committed:
Merge pull request #36 from derekxkwan/master

trivial commit to trapezoid~

porres committed:
updating help files

porres committed:
greaterthan~ object

porres committed:
delete greater than~ abstraction

porres committed:
more nettles

porres committed:
more nettles objects

porres committed:
new objects and help files for nettles

porres committed:
nettles update and new plus equals

porres committed:
rdiv rminus

porres committed:
new nettles help files and abstractions

DONE!!!!

flschiavoni committed:
Update equals.c

Added an alias creator ==~ to equals

porres committed:
Merge pull request #37 from flschiavoni/patch-1

Update equals.c

derekxkwan committed:
updating triangle~

derekxkwan committed:
some code cleaning in matrix~ mostly

derekxkwan committed:
adding updating authorship to matrix~

derekxkwan committed:
adding backup matrix to repo

derekxkwan committed:
fixing typos in cleaning up matrix~

porres committed:
Merge pull request #38 from derekxkwan/master

updating triangle~, some code cleaning in matrix~, scope~

porres committed:
trying alias in equals~

porres committed:
restoring triangle argument

in max is buggy, it is “0” when it should be 0.5, but I’m not gonna
make it “0” to be compatible with that bug because “0.5” makes total
sense.

porres committed:
updating output~ and help plus others

porres committed:
removing the build of scale until a release version comes

porres committed:
new asinh / atanh / acosh objects

testing

porres committed:
help for new math objects

porres committed:
Update README.md

derekxkwan committed:
changing float detection in args, updating matrix~ and deltaclip~

derekxkwan committed:
removing sickle/sic.c from includes in deltaclip~

porres committed:
Merge pull request #39 from derekxkwan/master

changing float detection in args, updating matrix~ and deltaclip~

porres committed:
creating new thresh object

porres committed:
trying to include new thresh~ object

working on this by having the code of Clip~ and sah~ as guides, merging
together structures from both of them. But there seems to be a conflict
and I can’t get it to compile. I’m getting the errors

“
sickle/thresh.c:29:23: error: use of undeclared identifier 'x'
    t_float lastout = x->x_lastout;
                      ^
sickle/thresh.c:39:5: error: use of undeclared identifier 'x'
    x->x_lastout = lastout;
    ^
”

but the same code in sah~ has no such issues.

porres committed:
seemingly fixed thresh code

porres committed:
updating thresh~ code and help, deleting abstraction

porres committed:
new biquad abstraction

porres committed:
updating help files of nettles and equals.c code

porres committed:
including equals2 - removing nettles abstractions, updating help for nettles objects and aliases, plus general help updates

porres committed:
fixing equals2 to load in extended and setting the ground for other objects

derekxkwan committed:
removing error argc<2 in matrix~

porres committed:
Merge pull request #40 from derekxkwan/master

removing error argc<2 in matrix~

porres committed:
including new scale code

porres committed:
help file for new scale object

porres committed:
updating help for scale

porres committed:
updating changelog to include alpha57 changes

porres committed:
text changes, trying removing line~ alias (worked)

porres committed:
restoring cyclone library

the original ready description described how cyclone could be set as a
regular lib. The -lib cyclone would call hammer and sickle objects plus
a small set of 12 objects with no alphanumerical names (old nettles.c
file).

Seems that in the Pd Extended days, the regular lib was split in
individual objects binaries for hammer/sickle but those 12 objects were
lost in this process.

I am now restoring the original cyclone library concept, to be included
in Pd’s startup, to load these 12 objects that were lost.

Recently, a “nettles” library was created to include these, but I
consider this a diversion from the original idea of having a cyclone
library, and I, for one, have never seen the word “nettles” and find it
hard, as a user, to relate it to cyclone…

This update successfully created a cyclone library that loads these 12
objects

porres committed:
renaming original cyclone library file

this file wasn’t being used and is now replaced by the new cyclone
library, containing the 12 non alphanumeric objects. So this original
file was renamed to cyclone_original

porres committed:
removing objects I was trying to rename to forbidden characters

porres committed:
changing Scope name, maintaining backwards compatibility

porres committed:
fixing scope name error

porres committed:
fixing scope again

porres committed:
final scope fix / cyclone help

porres committed:
add version

porres committed:
update scope in allSickles

porres committed:
bs

porres committed:
updating +=~ to Max 7

porres committed:
restoring original cyclone to cyclone.c

derekxkwan committed:
updating scope~

derekxkwan committed:
adding moved/updated scope~

porres committed:
Merge pull request #42 from derekxkwan/master

renaming/updating Scope

porres committed:
added signal input to main inlet

porres committed:
updating help of scope

porres committed:
added lastin

porres committed:
inserting comments on logic to include

porres committed:
some comments for flavio

porres committed:
more comments

porres committed:
some text changes

porres committed:
text changes, some cleanup and comments

porres committed:
name change / new all objects file

porres committed:
Update README.md

porres committed:
Update README.md

porres committed:
Update README.md

porres committed:
Update README.md

porres committed:
Update README.md

porres committed:
Update README.md

porres committed:
including original readme in current readme

porres committed:
including fred's ready

porres committed:
readme updates

porres committed:
ready is now About

porres committed:
restoring README

porres committed:
Update README.md

porres committed:
Update README.md

porres committed:
Update README.md

porres committed:
Update README.md

porres committed:
Update README.md

porres committed:
Update README.md

porres committed:
Update README.md

porres committed:
Update README.md

porres committed:
Update README.md

porres committed:
Update README.md

porres committed:
Update README.md

porres committed:
Update README.md

porres committed:
Update README.md

porres committed:
Update README.md

porres committed:
readme updates

porres committed:
Update README.md

porres committed:
Update README.md

porres committed:
Update README.md

porres committed:
Update README.md

porres committed:
including average2~ in the pool

porres committed:
Update README.md

porres committed:
Update README.md

porres committed:
Update README.md

porres committed:
Update README.md

porres committed:
Update README.md

porres committed:
Update README.md

porres committed:
Update README.md

porres committed:
Update README.md

porres committed:
Update README.md

porres committed:
more on texts and clean up for the day

porres committed:
add links

porres committed:
text edit
